### PR TITLE
Moons

### PR DESCRIPTION
**Pull request by AnandShegde**
## Description
Created notebook comparing the following techniques on Moons dataset.
* MLP
* Deep Ensemble
* MC Dropout
* Bootstrap
* NUTS
* GP
* SNGP
* VI

### Figures
image

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

### CODE REVIEW

**Comment by murphyk** [comment]
LGTM. One concern is that SNGP uses `DeepResNet(tf.keras.Model)` which is a different model to `cmlp`, so we're conflating two things here - the architecture and the estimation method. But changing this may be hard, so I will merge for now.
